File: SeekMatch-Server/SeekMatch.Core/Entities/Experience.cs
using SeekMatch.Core.Enums;

namespace SeekMatch.Core.Entities
{
    public class Experience : BaseEntity
    {
        public required string CompanyName { get; set; }
        public string? JobTitle { get; set; }
        public int StartYear { get; set; }
        public int StartMonth { get; set; }
        public int? EndYear { get; set; }
        public int? EndMonth { get; set; }
        public bool CurrentlyWorking { get; set; }
        public string? Description { get; set; }
        public ExperienceType Type { get; set; }
        public required Talent Talent { get; set; }
        public required string TalentId { get; set; }
    }
}

File: SeekMatch-Client/src/app/shared/models/experience.ts
import { JobType } from "../enums/enums";

export class Experience {

  public id: string | undefined = undefined;
  public companyName: string = '';
  public jobTitle: string = '';
  public startYear : number = 0;
  public startMonth : number = 0;
  public endYear : number = 0;
  public endMonth : number = 0;
  public currentlyWorking : boolean = false;
  public description: string = '';
  public type: JobType = JobType.FullTime;

  constructor(init?:Partial<Experience>) {
    Object.assign(this, init);
  }
}
